shared/XrUtility/XrExtensions.h

```diff
@@ -15,23 +15,37 @@
 //*********************************************************
 #pragma once
 
-#define FOR_EACH_EXTENSION_FUNCTION(_)           \
-    _(xrCreateSpatialAnchorMSFT)                 \
-    _(xrCreateSpatialAnchorSpaceMSFT)            \
-    _(xrDestroySpatialAnchorMSFT)                \
-    _(xrCreateHandTrackerEXT)                    \
-    _(xrDestroyHandTrackerEXT)                   \
-    _(xrLocateHandJointsEXT)                     \
-    _(xrCreateHandMeshSpaceMSFT)                 \
-    _(xrUpdateHandMeshMSFT)                      \
-    _(xrConvertWin32PerformanceCounterToTimeKHR) \
-    _(xrCreateSpatialGraphNodeSpaceMSFT)         \
-    _(xrGetD3D11GraphicsRequirementsKHR)         \
+#ifdef XR_USE_PLATFORM_WIN32
+#define FOR_EACH_WIN32_EXTENSION_FUNCTION(_) \
+    _(xrConvertWin32PerformanceCounterToTimeKHR)
+#else
+#define FOR_EACH_WIN32_EXTENSION_FUNCTION(_)
+#endif
+
+#ifdef XR_MSFT_CONTROLLER_MODEL_PREVIEW_EXTENSION_NAME
+#define FOR_EACH_CONTROLLER_MODEL_EXTENSION_FUNCTION(_) \
     _(xrGetControllerModelKeyMSFT)               \
     _(xrLoadControllerModelMSFT)                 \
     _(xrGetControllerModelPropertiesMSFT)        \
-    _(xrGetControllerModelStateMSFT)             \
-    _(xrGetVisibilityMaskKHR)
+    _(xrGetControllerModelStateMSFT)
+#else
+#define FOR_EACH_CONTROLLER_MODEL_EXTENSION_FUNCTION(_)
+#endif
+
+#define FOR_EACH_EXTENSION_FUNCTION(_)              \
+    _(xrCreateSpatialAnchorMSFT)                    \
+    _(xrCreateSpatialAnchorSpaceMSFT)               \
+    _(xrDestroySpatialAnchorMSFT)                   \
+    _(xrCreateHandTrackerEXT)                       \
+    _(xrDestroyHandTrackerEXT)                      \
+    _(xrLocateHandJointsEXT)                        \
+    _(xrCreateHandMeshSpaceMSFT)                    \
+    _(xrUpdateHandMeshMSFT)                         \
+    _(xrCreateSpatialGraphNodeSpaceMSFT)            \
+    _(xrGetD3D11GraphicsRequirementsKHR)            \
```

Comment:
xrGetD3D11GraphicsRequirementsKHR should also go into a separate FOR_EACH based on XR_USE_GRAPHICS_API_D3D11 being defined.

Reply:
This change is heading to a good direction.  We can have follow ups to break all extension functions into smaller groups.  d3d11 is one of them, anything in this list that's not core should be broken down.   Not necessarily in this change though.

Reply:
its pretty easy to add this change. I will say that i am not sure how to test scenarios when d3d11 is not available

Reply:
The CI will make sure the build still works, so I think that is sufficient testing.